test/Microsoft.NET.Build.Tests/GivenThatWeWantToGenerateADepsFileForATool.cs

```diff
@@ -0,0 +1,239 @@
+﻿using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using NuGet.Packaging;
+using NuGet.ProjectModel;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Xml.Linq;
+using Xunit;
+using static Microsoft.NET.TestFramework.Commands.MSBuildTest;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToGenerateADepsFileForATool : SdkTest
+    {
+        [Fact]
+        public void It_creates_a_deps_file_for_the_tool_and_the_tool_runs()
+        {
+            TestProject toolProject = new TestProject()
+            {
+                Name = "TestTool",
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.0",
+                IsExe = true
+            };
+
+            GenerateDepsAndRunTool(toolProject)
+                .Should()
+                .Pass()
+                .And.HaveStdOutContaining("Hello World!");
+        }
+
+        [Fact]
+        public void It_handles_conflicts_when_creating_a_tool_deps_file()
+        {
+            TestProject toolProject = new TestProject()
+            {
+                Name = "DependencyContextTool",
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.0",
+                IsExe = true
+            };
+
+            toolProject.PackageReferences.Add(new TestPackageReference("Microsoft.Extensions.DependencyModel", "2.0.0-preview1-002022", null));
+
+            string toolSource = @"
+using System;
+using System.Linq;
+using Microsoft.Extensions.DependencyModel;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        if(DependencyContext.Default?.RuntimeGraph?.Any() == true)
+        {
+            Console.WriteLine(""Successfully loaded runtime graph"");
+        }
+        else
+        {
+            Console.WriteLine(""Couldn't load runtime graph"");
+        }
+    }
+}";
+
+            toolProject.SourceFiles.Add("Program.cs", toolSource);
+
+            GenerateDepsAndRunTool(toolProject, "ToolConflictResolution")
+                .Should()
+                .Pass()
+                .And.HaveStdOutContaining("Successfully loaded runtime graph");
+        }
+
+        //  This method duplicates a lot of logic from the CLI in order to test generating deps files for tools in the SDK repo
```

Comment:
I wonder if we need to go this far?

Could we get away with just testing the GenerateDeps.proj file without bringing a bunch of "tools" stuff.  Just get a project.assets.json file, and invoke the GenerateDeps.proj file on it....

thoughts?